standard/ERCs/erc-6900.md

```diff
@@ -471,72 +471,55 @@ An account can have the same validation module installed more than once.
 
-The function MUST record the manifest hash and dependencies that were used for the module's installation. Each dependency's record MUST also be updated to reflect that it has a new dependent. These records MUST be used to ensure calls to `uninstallModule` are comprehensive and undo all edited configuration state from installation. The mechanism by which these records are stored and validated is up to the implementation.
+During validation uninstallation, the account MUST correctly clear flags and other fields based on the incoming data provided by users.
 
-The function MUST store the module's permitted function selectors, permitted external calls, and whether it can spend the account's native tokens, to be able to validate calls to `executeFromModule` and `executeFromModuleExternal`.
+- the account MUST clear all flags for the validation function, including `isGlobal` and `isSignatureValidation`.
+- the account MUST remomve all hooks and clear hook module states by calling `onUninstall` with the user-provided data for each hook, including both pre validation hooks and permission hooks.
+- the account MUST remove all selectors that the validation function can validate.
+- the account MUST emit `ValidationUninstalled` as defined in the interface for all uninstalled validations.
 
-The function MUST parse through the execution functions, validation functions, and hooks in the manifest and add them to the modular account after resolving each `ManifestFunction` type.
+#### Execution and their installation /uninstallation
 
-- Each execution function selector MUST be added as a valid execution function on the modular account. If the function selector has already been added or matches the selector of a native function, the function SHOULD revert.
-- If a validation function is to be added to a selector that already has that type of validation function, the function SHOULD revert.
+An account can install any number of execution functions.
+An execution function selector MUST be unique to the account.
+An execution function selector MUST not conflict with native ERC-4337 and ERC-6900 functions.
 
-The function MAY store the interface IDs provided in the manifest's `interfaceIds` and update its `supportsInterface` behavior accordingly.
+During execution installation, the account MUST correctly set flags and other fields based on the incoming data and module manifest provided by users.
 
-Next, the function MUST call the module's `onInstall` callback with the data provided in the `moduleInstallData` parameter. This serves to initialize the module state for the modular account. If `onInstall` reverts, the `installModule` function MUST revert.
+- the account MUST install all execution functions and set flags and fields accordingly as specified in the manifest.
+- the account MUST add all execution hooks as specified in the manifest.
+- the account MUST add all supported interfaces as specified in the manifest.
```

Comment:
I think we've marked supporting ERC-165 on the account itself as optional, so we should change the "MUST" here to "SHOULD".

Reply:
hmm, I could not find wording for that. All ref to 165 are either `requires`, `MUST`. If the convo happened offline, can you share? I might have missed it.

Reply:
It was optional in the previous spec (v0.7). One of the lines removed in this PR is:

> The function MAY store the interface IDs provided in the manifest's `interfaceIds` and update its `supportsInterface` behavior accordingly.

So it's something we haven't changed since v0.7. We could update it to be required in v0.8, but we haven't really discussed that much. I'd lean towards keeping it optional.

(This is about the account having its own `supportsInterface` function, it is required for the modules).

Reply:
Sounds good. wording should be updated.